include/astarte_device.h

```diff
@@ -349,6 +349,14 @@ astarte_err_t astarte_device_unset_path(astarte_device_handle_t device, const ch
  */
 bool astarte_device_is_connected(astarte_device_handle_t device);
 
+/**
+ * @brief Get the encoded hardware ID of the device.
+ *
+ * @details Get the encoded hardware ID of the device.
+ * @param device An Astarte device handle.
+ * @return The string containing the encoded device ID.
```

Comment:
we should document that the string is owned by astarte_device_handle_t and it will be freed as soon as the device is closed / freed.